srcs/request_handling/GetHandling.cpp

```
@@ -2,196 +2,207 @@
-          << "</html>";
-
-    return index.str();
+	if (has_file_ending(path))
+	{
+		auto index_file = load_file_with_cache(path); // try to open index if available
+		if (index_file.has_value())
+			return index_file.value().content;
+
+		size_t last_slash = path.find_last_of('/'); // remove file ending
+		if (last_slash != std::string::npos)
+			path = path.substr(0, last_slash);
```

cppcheck

[performance] Ineffective call of function 'substr' because a prefix of the string is assigned to itself. Use resize() or pop_back() instead.

(uselessCallsSubstr)